src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Store.cs

```diff
@@ -103,6 +135,33 @@ public void Add(X509Certificate2 certificate)
             _storePal.Add(certificate.Pal);
         }
 
+        public void AddRange(X509Certificate2Collection certificates)
+        {
+            if (certificates == null)
+                throw new ArgumentNullException(nameof(certificates));
+
+            int i = 0;
+            try
+            {
+                foreach (X509Certificate2 certificate in certificates)
+                {
+                    Add(certificate);
+                    i++;
+                }
+            }
+            catch
+            {
+                // For desktop compat, we keep the exception semantics even though they are not ideal
+                // because a duplicate will raise exception causing certs to be removed even if 
+                // they weren't there before.
+                for (int j = 0; j < i; j++)
```

Comment:
Is the enumeration order with the collection's GetEnumerator guaranteed to be the same as indexing from 0 to N? (I assume so, just verifying.)